src/main/java/net/mcreator/generator/template/TemplateExpressionParser.java

```diff
@@ -123,17 +97,41 @@ private static boolean parseCondition(@Nonnull Generator generator, @Nonnull Str
 		return false;
 	}
 
-	private static boolean processFTLExpression(Generator generator, String expression) {
+	private static Object getValueFrom(String field, Object conditionDataProvider) throws ReflectiveOperationException {
+		if (!field.contains("()")) { // field
+			return conditionDataProvider.getClass().getField(field.trim()).get(conditionDataProvider);
+		} else { // method
+			return conditionDataProvider.getClass().getMethod(field.replace("()", "").trim())
+					.invoke(conditionDataProvider);
+		}
+	}
+
+	public static Object processFTLExpression(Generator generator, String expression, Object dataHolder) {
 		try {
-			Template t = new Template("INLINE EXPRESSION", new StringReader(expression),
+			Map<String, Object> dataModel = new HashMap<>(generator.getBaseDataModelProvider().provide());
+			AtomicReference<?> retVal = new AtomicReference<>(null);
+			dataModel.put("retVal", retVal);
+			if (dataHolder != null)
+				dataModel.put("data", dataHolder);
+
+			String expr = !checkStartsWithAnyKey(expression, dataModel) && dataHolder != null ?
+					"data." + expression : // by default, dataHolder is used to process the expression
+					expression;
+			Template t = new Template("INLINE EXPRESSION", new StringReader("${retVal.set(" + expr + ")}"),
 					generator.getGeneratorConfiguration().getTemplateGenConfigFromName("templates").getConfiguration());
+			t.process(dataModel, new StringWriter());
 
-			StringWriter stringWriter = new StringWriter();
-			t.process(generator.getBaseDataModelProvider().provide(), stringWriter);
-
-			return Boolean.parseBoolean(stringWriter.getBuffer().toString());
+			return retVal.get();
 		} catch (Exception e) {
 			LOG.error("Failed to parse FTL expression: " + expression, e);
+			return null;
+		}
+	}
+
+	private static boolean checkStartsWithAnyKey(String expression, Map<String, Object> dataModel) {

```

Comment:
Add JavaDOC here, describing what this method does and why is is needed

Reply:
:heavy_check_mark: Fixed by removing the method